jvm/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java

```diff
@@ -78,6 +84,43 @@ public static StringListNode makeStringList(ArrayList<String> strConstants) {
         return new StringListNode(strConstants);
     }
 
+    public static ExpressionNode makeLiteral(Object obj, DataType dataType, Boolean nullable) {
+        if (dataType instanceof IntegerType) {
+            if (obj == null) {
+                return makeNullLiteral(TypeBuilder.makeI32(nullable));
+            } else {
+                return makeIntLiteral((Integer) obj);
+            }
+        } else if (dataType instanceof LongType) {
+            if (obj == null) {
+                return makeNullLiteral(TypeBuilder.makeI64(nullable));
+            } else {
+                return makeLongLiteral((Long) obj);
+            }
+        } else if (dataType instanceof DoubleType) {
+            if (obj == null) {
+                return makeNullLiteral(TypeBuilder.makeFP64(nullable));
+            } else {
+                return makeDoubleLiteral((Double) obj);
+            }
+        } else if (dataType instanceof DateType) {
+            if (obj == null) {
+                return makeNullLiteral(TypeBuilder.makeDate(nullable));
+            } else {
+                return makeDateLiteral((Integer) obj);
+            }
+        } else if (dataType instanceof StringType) {
+            if (obj == null) {
+                return makeNullLiteral(TypeBuilder.makeString(nullable));
+            } else {
+                return makeStringLiteral((String) obj);
```

Comment:
Use `obj.toString()` here, when obj is StringType, in fact it's `UTF8String`, can not use `(String) obj` to convert UTF8String to String.